CMakeLists.txt

```diff
@@ -182,6 +182,13 @@ endif()
 
 if (MSVC)
   find_package(PThreads4W REQUIRED)
+  #find library from vcpkg gettimeofday
+    find_library(GETTIMEOFDAY_LIBRARY
+          NAMES gettimeofday
+  )
+    if (NOT GETTIMEOFDAY_LIBRARY)
+      message(FATAL_ERROR "Could not find gettimeofday.lib")
+    endif()
```

Comment:
Given that this is vcpkg specific and we want to support other environments/package managers such as KDE Craft too, I'd prefer we include the files from https://github.com/microsoft/vcpkg/tree/7cfd63db7ff8855ab8e2baebeb80b37f5c0818db/ports/gettimeofday ourself. Since it consists of very few small files it seems okay to just  copy them `src/msvc` in my opinion.